File: tests/fixtures/python/primitive_obsession.py
"""Functions heavy on primitive types without domain abstractions (future)."""


def create_invoice(
    customer_name: str,
    customer_email: str,
    customer_phone: str,
    customer_address: str,
    item_name: str,
    item_price: float,
    item_quantity: int,
    tax_rate: float,
    discount_code: str,
    shipping_method: str,
    notes: str,
):
    subtotal = item_price * item_quantity
    tax = subtotal * tax_rate
    total = subtotal + tax
    return {
        "customer": customer_name,
        "email": customer_email,
        "total": total,
    }


def send_notification(
    recipient_name: str,
    recipient_email: str,
    subject: str,
    body: str,
    priority: str,
    template: str,
    reply_to: str,
):
    pass

File: tests/fixtures/csharp/primitive_obsession.cs
public class InvoiceService
{
    public object CreateInvoice(string customerName, string customerEmail,
                                string customerPhone, string customerAddress,
                                string itemName, double itemPrice,
                                int itemQuantity, double taxRate,
                                string discountCode, string shippingMethod,
                                string notes)
    {
        double subtotal = itemPrice * itemQuantity;
        double tax = subtotal * taxRate;
        double total = subtotal + tax;
        return new { customer = customerName, email = customerEmail, total };
    }

    public void SendNotification(string recipientName, string recipientEmail,
                                 string subject, string body,
                                 string priority, string template,
                                 string replyTo)
    {
    }
}
